cirq/ops/pauli_gates.py

```diff
@@ -52,7 +52,7 @@ def _commutes_(self, other: Any,
                    atol: float) -> Union[bool, NotImplementedType, None]:
         if not isinstance(other, Pauli):
             return NotImplemented
-        return self is other
+        return self == other
```

Comment:
I think that rather than making this work with multiple Pauli instances, we could instead change `_Pauli{X,Y,Z}._from_json_dict_` to return one of the three instances stored in `Pauli._XYZ`. This way we always have a single instance of each Pauli and the `is` operator` is fine. This is also consistent with how we do things in methods like `by_index` and `third`.

Is there a reason this wouldn't work?

(One obstacle to this is that the internal classes `_Pauli{X,Y,Z}` take the exponent argument. I think they shouldn't and filed #2920 about this.)

Reply:
Now that you have fixed #2920, could you restore this line to `self is other` and instead modify `_Pauli{X,Y,Z}._from_json_dict_` to return the appropriate Pauli instance from `Pauli._XYZ`? This should fix #2813.

Reply:
Done. Also added  a test which fails before this change.